typing/path.ml

```diff
@@ -22,21 +22,27 @@ and extra_ty =
   | Pcstr_ty of string
   | Pext_ty
 
-let rec same p1 p2 =
-  p1 == p2
-  || match (p1, p2) with
-    (Pident id1, Pident id2) -> Ident.same id1 id2
-  | (Pdot(p1, s1), Pdot(p2, s2)) ->
-      s1 = s2 && same p1 p2
-  | (Papply(fun1, arg1), Papply(fun2, arg2)) ->
-      same fun1 fun2 && same arg1 arg2
-  | (Pextra_ty (p1, t1), Pextra_ty (p2, t2)) ->
-      let same_extra = match t1, t2 with
-        | (Pcstr_ty s1, Pcstr_ty s2) -> s1 = s2
-        | (Pext_ty, Pext_ty) -> true
-        | ((Pcstr_ty _ | Pext_ty), _) -> false
-      in same_extra && same p1 p2
-  | (_, _) -> false
+let same_aux ident_cmp p1 p2 =
+  let rec aux p1 p2 =
```

Comment:
For future reference, another way to do this without introducing an auxiliary recursive function:

```ocaml
let rec same ident_cmp p1 p2 =
  let same p1 p2 = same ident_cmp p1 p2 in
  ...
```